source/vibe/db/postgresql/package.d

```diff
@@ -107,16 +107,32 @@ class __Conn : dpq2.Connection
     {
         import vibe.core.core;
 
+        // Connection socket should be non-blocking while waiting
```

Comment:
Yes, but AFAIK it is already in non blocking state, so there is no need to setup it to non blocking again, and definitly it is wrong to setup it to blocking

Reply:
> but AFAIK it is already in non blocking state

Nope. Do not confuse non-blocking socket with non-blocking state of Postgres connection

Reply:
Yes it is, you can try it if you do not beleive me ;-)

Reply:
So I looked at PQconnectdb(https://github.com/postgres/postgres/blame/5c15a54e851ecdd2b53e6d6a84f8ec0802ffc3cb/src/interfaces/libpq/fe-connect.c#L592) source code (libpq) and it is implemented atop of  PQconnectStart which setup socket as non-blocking

Reply:
Isn't see. Can you cite code that enables non-block on the socket?

Reply:
https://github.com/postgres/postgres/blob/5c15a54e851ecdd2b53e6d6a84f8ec0802ffc3cb/src/interfaces/libpq/fe-connect.c#L2135